src/runtime_src/core/edge/user/hwctx_object.cpp

```
@@ -60,4 +81,15 @@ namespace zynqaie {
     return nullptr;
 #endif
   }
+
+  std::unique_ptr<xrt_core::profile_handle>
+  hwctx_object::open_profile_handle()
+  {
+#ifdef XRT_ENABLE_AIE    
+    return std::make_unique<profile_object>(m_shim, m_aie_array.get());
+#else
+    return nullptr;
```

throw an exception instead of returning nullptr